tool/embedded-binaries.ts

```diff
@@ -2,59 +2,103 @@
 // MIT-style license that can be found in the LICENSE file or at
 // https://opensource.org/licenses/MIT.
 
-import * as p from 'path';
-import {promises as fs, existsSync} from 'fs';
-import fetch, {RequestInit} from 'node-fetch';
-import {extract as extractTar} from 'tar';
 import extractZip = require('extract-zip');
+import {promises as fs, existsSync} from 'fs';
+import fetch from 'node-fetch';
+import * as p from 'path';
+import {satisfies} from 'semver';
 import * as shell from 'shelljs';
+import {extract as extractTar} from 'tar';
 
 shell.config.fatal = true;
 
+// The current platform's operating system. Throws if the operating system
+// is not supported by Dart Sass Embedded.
+let OS: 'linux' | 'macos' | 'windows';
+switch (process.platform) {
+  case 'linux':
+    OS = 'linux';
+    break;
+  case 'darwin':
+    OS = 'macos';
+    break;
+  case 'win32':
+    OS = 'windows';
+    break;
+  default:
+    throw Error(`Platform ${process.platform} is not supported.`);
+}
```

Comment:
```suggestion
const OS: 'linux' | 'macos' | 'windows' = (() => {
  switch (process.platform) {
   case 'linux':
      return 'linux';
    case 'darwin':
      return 'macos';
    case 'win32':
      return 'windows';
    default:
      throw Error(`Platform ${process.platform} is not supported.`);
  }
})();
```

This will let you declare these as `const` (since they are logically constants) while also having complex initializer logic.

Reply:
Good call.